### PR TITLE
JS Components: add the new subTitle prop to the ProductOffer component

### PR DESCRIPTION
**Pull request by retrofox**
This PR adds a new `subTitle` property that is rendered just below to the primary title of the ProductOffer component.

Fixes #

#### Changes proposed in this Pull Request:

* JS Components: add the new subTitle prop to the ProductOffer component

#### Testing instructions:

* Use Storybook
* Go to the ProductOffer component:
`http://localhost:50240/?path=/docs/js-packages-components-product-offer--security-bundle`
* Set a subTitle value using the story controls
* Take a look at the spacing values, font-size, etc.

https://user-images.githubusercontent.com/77539/163850599-f4cb371b-fc30-4017-8bea-fe9edf202db5.mov

### CODE REVIEW

**Review by leogermani — APPROVED** [review]
LGTM

File: projects/js-packages/components/components/product-offer/index.jsx [hunk 1]
```diff
@@ -123,7 +125,8 @@ const ProductOffer = ( {
 					products={ supportedProducts?.length ? supportedProducts : [ slug ] }
 				/>
 				<H3>{ title }</H3>
-				<Text mb={ 3 }>{ description }</Text>
+				{ subTitle && <Title mb={ 3 }>{ subTitle }</Title> }
```
Comment (renatoagds) [inline projects/js-packages/components/components/product-offer/index.jsx:128]: For me looks really strange use a component `Title` with `subTitle`, but don't think we have a better approach for that, just adding the discussion for more ideas (if someone has).

**Review by renatoagds — APPROVED** [review]
🚀

**Comment by leogermani** [comment]
r243950-wpcom

File: projects/js-packages/components/components/product-offer/index.jsx (hunk 1 above)
Comment (retrofox) [reply projects/js-packages/components/components/product-offer/index.jsx:128]: Same thoughts.
Probably we should consider to changing component names and or properties names.
These changes are based on the design:

**Pull request merged**